src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -128,7 +128,8 @@ OpResult CondTraverseReset(OpBase *ctx) {
 void CondTraverseFree(OpBase *ctx) {
     CondTraverse *op = (CondTraverse*)ctx;
     if(op->iter) TuplesIter_free(op->iter);
-    if(op->F) GrB_Matrix_free(&op->F);
+    // TODO It seems that this should be a safe free, but it causes a lot of errors in parallel queries.
+    // if(op->F) GrB_Matrix_free(&op->F);
```

Comment:
I really don't think that this commenting should be necessary, but I reliably segfaulted with it uncommented outside of Valgrind (no errors reported within Valgrind).

This most likely indicates another race condition, which I will keep looking for!

Reply:
So, currently we're leaking here?